pkg/worker/pool.go

```diff
@@ -20,34 +21,49 @@ func NewWorkerPool() *WorkerPool {
 	return &WorkerPool{workers: make([]Worker, 0)}
 }
 
-// StartWorkers cycles through all the workers
+// Start cycles through all the workers
 // currently inside the WorkerPool and creates
 // a goroutine for each. The 'Start' method of
 // each worker is executed concurrently.
-func (pool *WorkerPool) StartWorkers() {
+//
+// Start does NOT block, however consumers
+// can wait on the WaitGroup in the pool if they
+// wish.
+func (pool *WorkerPool) Start() {
+	if pool.started {
+		panic("cannot start an already started worker pool")
+	}
+
+	pool.started = true
 	for _, worker := range pool.workers {
```

Comment:
I'm wondering if a lot of this logic could be handled by a [sync/errgroup](https://pkg.go.dev/golang.org/x/sync/errgroup)

Reply:
Honestly I had a look at this package and couldn't quite understand how I'd use it. Perhaps we can have a chat about it :3